experimental/dds/tree2/src/core/edit-manager/editManager.ts

```diff
@@ -237,19 +242,21 @@ export class EditManager<
 		newCommit: Commit<TChangeset>,
 		sequenceNumber: SeqNumber,
 		referenceSequenceNumber: SeqNumber,
+		undoRedoManager: UndoRedoManager<TChangeset, any>,
 	): Delta.Root {
 		if (newCommit.sessionId === this.localSessionId) {
 			// `newCommit` should correspond to the oldest change in `localChanges`, so we move it into trunk.
 			// `localChanges` are already rebased to the trunk, so we can use the stored change instead of rebasing the
 			// change in the incoming commit.
 			const localPath = getPathFromBase(this.localBranch, this.trunk);
 			// Get the first revision in the local branch, and then remove it
-			const { change } =
+			const commit =
 				localPath.shift() ??
 				fail(
 					"Received a sequenced change from the local session despite having no local changes",
 				);
-			this.pushToTrunk(sequenceNumber, { ...newCommit, change });
+			this.trunkUndoRedoManager.trackCommit(commit);
```

Comment:
Oh that's a great call! I fixed this without moving the `trackCommit` into `pushToTrunk` because only commits that were local should be tracked. It still didn't fixed the findCommonAncestor call though 😞.

Reply:
Ohhh I see.